pyingest/parsers/jats.py

```diff
@@ -216,27 +218,38 @@ def parse(self, fp, **kwargs):
                     pass
 
                 # Author names
-                if a.find('surname') is not None:
-                    surname = self._detag(a.surname, [])
-                else:
-                    surname = 'Anonymous'
-                if a.find('prefix') is not None:
-                    prefix = self._detag(a.prefix, []) + ' '
-                else:
-                    prefix = ''
-                if a.find('suffix') is not None:
-                    suffix = ' ' + self._detag(a.suffix, [])
-                else:
-                    suffix = ''
-                if a.find('given-names') is not None:
-                    given = self._detag(a.find('given-names'), [])
+                if a.find('collab') is not None:
+                   base_metadata['authors'].append(self._detag(a.collab, []))
                 else:
-                    given = ''
-                forename = prefix + given + suffix
-                if forename == '':
-                    base_metadata['authors'].append(surname)
-                else:
-                    base_metadata['authors'].append(surname + ', ' + forename)
+                    if a.find('surname') is not None:
+                        surname = self._detag(a.surname, [])
+                    else:
+#                   surname = 'Anonymous'
+                        # print "Warning: Undefined surname: ",a
+                        surname = ''
+                    if a.find('prefix') is not None:
+                        prefix = self._detag(a.prefix, []) + ' '
+                    else:
+                        prefix = ''
+                    if a.find('suffix') is not None:
+                        suffix = ' ' + self._detag(a.suffix, [])
+                    else:
+                        suffix = ''
+                    if a.find('given-names') is not None:
+                        given = self._detag(a.find('given-names'), [])
+                    else:
+                        given = ''
+                    forename = prefix + given + suffix
+                    if forename == '':
+                        if surname != '':
+                            base_metadata['authors'].append(surname)
+                        else:
+                            base_metadata['authors'].append('ANONYMOUS')
```

Comment:
This looks new to me (assigning "ANONYMOUS" when an author name is not given).  I'd ask Carolyn if this is what we really want, since usually we'd just leave the field empty when author info is not known

Reply:
This was removed.  I needed to make changes to the author_names module to deal with cases where an author XML entity wound up being NoneType.  This shouldn't happen for valid XML data unless the publisher provides an XML author field with no data.